book/src/contributing/architecture.md

```diff
@@ -0,0 +1,100 @@
+# Architecture
+
+This document describes implementation details of this library.
+
+## Terminology
+
+**Rust Wire** types refers to the C types that the Dart VM uses to communicate with the Rust library.
+
+**Dart Wire** types are the Dart counterpart of Rust wire types, but in
+the `*.io.dart` files. Both Rust and Dart wire types communicate using the
+vocabulary of C types, aka primitives, structs, unions and pointers.
+
+**Rust JS** (also WASM) types are the WASM equivalent of Rust wire
```

Comment:
To make terms consistent (we have already said "wire" and "api" types), maybe we can call the terms in this doc "Rust Wire" -> "Rust Wire IO", and "Rust JS" -> "Rust Wire JS"?